### PR TITLE
Add to the networking module header a definition of the purpose and use of the module

### PR DESCRIPTION
**Pull request by sbutcher-arm**
## Description
The purpose of the networking module can sometimes be misunderstood. This adds a definition and explanation of what the networking module is and what it can be used for.

This PR changes the doxygen documentation only. 

## Status
**READY**

## Requires Backporting
NO  

Whilst the change only clarifies the purpose of the networking module, it is restrictive, and it seems unnecessary to retrospectively say what the module can or can't be used for on maintained branches.

### CODE REVIEW

File: include/mbedtls/net_sockets.h [hunk 1]
```diff
@@ -1,7 +1,22 @@
 /**
  * \file net_sockets.h
  *
- * \brief Network communication functions
+ * \brief   Network sockets abstraction layer to integrate Mbed TLS into a
+ *          BSD-style sockets API.
+ *
+ *          The network sockets module provides an example integration of the
+ *          Mbed TLS library into a BSD sockets implementation. The module is
```
Comment (RonEld) [inline include/mbedtls/net_sockets.h]: Although I am not a native English speaker, I agree with @irwir 's comment on using "both" and "also".
I would remove the word "both" and replace "and also" with "as well as"

**Comment by RonEld** [comment]
@irwir
>It is somewhat unclear if a third part application could use networking module as a part of the Mbed TLS library, or not.

 I believe the sentence is meant to be that the networking module is meant to be used by Mbed TLS only, and not meant to be a standalone networking module, for any other library. Do you have another suggestion for this sentence?

**Comment by irwir** [comment]
> Do you have another suggestion for this sentence?

This is my understanding of the general idea.

> The module is intended to be an example of how Mbed TLS can be integrated into a networking stack, and also act as Mbed TLS's integration on the supported platforms,.
> The module is not intended to be used by third party application software without Mbed TLS's integration.

**Comment by sbutcher-arm** [comment]
@RonEld / @irwir - I've revised the language to try and make it clearer. Please let me know what you think.

**Comment by irwir** [comment]
That was clearer.
While on the subject of grammar, could someone take care of this strange [indefinite article](https://github.com/ARMmbed/mbedtls/blob/7d728bd70e6a8d129ee07813da1538de1bf95cb2/include/mbedtls/config.h#L1631)?

**Pull request merged**